annotation_widgets/event_validation/gui.py

```diff
@@ -197,11 +188,67 @@ def apply_color(self, question, selected_answer):
                         rb.config(bg=self.questions_map[question][selected_answer])
 
 
+class VideoFrameSlider(tk.Frame):
+    def __init__(self, parent, from_, to, callback=None, *args, **kwargs):
+        super().__init__(parent, *args, **kwargs)
+
+        self.callback = callback
+        self.is_playing = False
+
+        button_frame = tk.Frame(self)
+        button_frame.pack(side="left", padx=10, pady=5, anchor="s")
+
+        # Play/Pause Button
+        self.play_button = tk.Button(button_frame, text="▶", width=3, font=("Arial", 10), command=self.toggle_play_pause)
+        self.play_button.pack(side="left", padx=(0, 5), anchor="s")
+
+        # Stop Button
+        self.stop_button = tk.Button(button_frame, text="■", width=3, font=("Arial", 10), command=self.stop)
+        self.stop_button.pack(side="left", padx=(5, 0), anchor="s")
+
+        # Slider
+        self.slider = tk.Scale(self, from_=from_, to=to, orient="horizontal", command=self.on_slider_change, length=500, sliderlength=30, width=20)
+        self.slider.pack(side="right", fill="x", expand=True, padx=(10, 10), anchor="n")
+
+
+    def on_slider_change(self, value):
+        if self.callback is not None:
+            self.callback(value)
+
+    def toggle_play_pause(self):
+        self.is_playing = not self.is_playing
+
+        if self.is_playing:
+            self.play()
+        else:
+            self.pause()
+        self.master.handle_play_pause(self.is_playing)
+
+    def play(self):
+        self.is_playing = True
+        self.play_button.config(text="II")
+
+    def pause(self):
+        self.is_playing = False
+        self.play_button.config(text="▶")
+
+    def stop(self):
+        self.is_playing = False
+        self.pause()
+        self.master.handle_stop()
+
+    def show(self):
+        self.grid()
+
+    def hide(self):
+        self.grid_remove()
+
+
 class BaseCanvasView(tk.Canvas):
     """
     Class Contains base functionality of displaying Images in window.
     """
-    def __init__(self, parent: tk.Tk, root: tk.Tk, logic: AbstractImageAnnotationLogic):
+    def __init__(self, parent: tk.Tk, root: tk.Tk, logic: AbstractImageAnnotationLogic): # Don't pass logic class here, use only callbacks to set or obtain values.
```

Comment:
Don't pass logic class here, use only callbacks to set or obtain values.

Reply:
Refactored with callbacks